iset.mm

```diff
@@ -117423,8 +117486,619 @@ Infinite sums (cont.)
+      -> ( tan ` A ) = ( ( ( exp ` ( 2 x. ( _i x. A ) ) ) - 1 ) /
+      ( _i x. ( ( exp ` ( 2 x. ( _i x. A ) ) ) + 1 ) ) ) ) $=
+    ( cc wcel c2 ci cmul co ce cfv c1 caddc cc0 cap wbr cmin ax-icn sylancr syl
+    cdiv wceq wa cneg ctan simpl mulcl efcl negicn subcld cexp cz efexp sylancl
+    addcld 2z sqvald eqtrd mulneg1 fveq2d oveq2d efcan eqtr2d adddid eqtr4d a1i
+    oveq12d mul12d 2cn ax-1cn addcl iap0 simpr mulap0d eqbrtrrd mulap0bbd efap0
+    divcanap5d subdid ccos cosval adantr 2cnd divap0d tanval2ap syldan 3eqtr4rd
+    2ap0 eqbrtrd ) ABCZDEAFGZFGZHIZJKGZLMNZUAZWIHIZWOEUBZAFGZHIZOGZFGZWOEWOWRKG
+    ZFGZFGZSGWSXBSGZWKJOGZEWLFGZSGAUCIZWNWSXBWOWNWOWRWNWIBCZWOBCWNEBCZWHXHPWHWM
+    UDZEAUEQZWIUFRZWNWQBCZWRBCWNWPBCWHXMUGXJWPAUEQWQUFRZUHWNXIXABCXBBCPWNWOWRXL
+    XNUMZEXAUEQZXLWNWOXBXLXPWNXFXCLMWNXFEWOXAFGZFGXCWNWLXQEFWNWLWOWOFGZWOWRFGZK
+    GXQWNWKXRJXSKWNWKWODUIGZXRWNXHDUJCWKXTTXKUNWIDUKULWNWOXLUOUPZWNXSWOWIUBZHIZ
+    FGZJWNWRYCWOFWNWQYBHWNXIWHWQYBTPXJEAUQQURUSWNXHYDJTXKWIUTRVAZVEWNWOWOWRXLXL
+    XNVBVCUSWNEWOXAXIWNPVDZXLXOVFUPZWNEWLYFWNWKBCZJBCWLBCWNWJBCZYHWNDBCXHYIVGXK
+    DWIUEQWJUFRVHWKJVIULELMNWNVJVDWHWMVKVLVMVNZWNXHWOLMNXKWIVORVPWNXEWTXFXCSWNX
+    EXRXSOGWTWNWKXRJXSOYAYEVEWNWOWOWRXLXLXNVQVCYGVEWHWMAVRIZLMNXGXDTWNYKXADSGZL
+    MWHYKYLTWMAVSVTWNXADXOWNWAWNEXAYFXOYJVNDLMNWNWFVDWBWGAWCWDWE $.
+
+  $( The sine of a real number in terms of the exponential function.
+     (Contributed by NM, 30-Apr-2005.) $)
+  resinval $p |- ( A e. RR -> ( sin ` A ) = ( Im ` ( exp ` ( _i x. A ) ) ) ) $=
+    ( cr wcel ci cmul co ce cfv cneg cmin c2 cdiv csin cc ax-icn sylancr oveq2d
+    ccj wceq syl cim recn cjmul cji oveq1i cjre syl5eq eqtrd fveq2d efcj eqtr3d
+    mulcl oveq1d sinval efcl imval2 3syl 3eqtr4d ) ABCZDAEFZGHZDIZAEFZGHZJFZKDE
+    FZLFZVAVARHZJFZVFLFZAMHZVAUAHZUSVEVIVFLUSVDVHVAJUSUTRHZGHZVDVHUSVMVCGUSVMDR
+    HZARHZEFZVCUSDNCZANCZVMVQSOAUBZDAUCPUSVQVBVPEFVCVOVBVPEUDUEUSVPAVBEAUFQUGUH
+    UIUSUTNCZVNVHSUSVRVSWAOVTDAULPZUTUJTUKQUMUSVSVKVGSVTAUNTUSWAVANCVLVJSWBUTUO
+    VAUPUQUR $.
+
+  $( The cosine of a real number in terms of the exponential function.
+     (Contributed by NM, 30-Apr-2005.) $)
+  recosval $p |- ( A e. RR -> ( cos ` A ) = ( Re ` ( exp ` ( _i x. A ) ) ) ) $=
+    ( cr wcel ci cmul co ce cfv cneg caddc c2 cdiv ccj ccos wceq ax-icn sylancr
+    cc oveq2d syl cre recn cjmul cji oveq1i cjre syl5eq eqtrd fveq2d mulcl efcj
+    eqtr3d oveq1d cosval efcl reval 3syl 3eqtr4d ) ABCZDAEFZGHZDIZAEFZGHZJFZKLF
+    ZVAVAMHZJFZKLFZANHZVAUAHZUSVEVHKLUSVDVGVAJUSUTMHZGHZVDVGUSVLVCGUSVLDMHZAMHZ
+    EFZVCUSDRCZARCZVLVPOPAUBZDAUCQUSVPVBVOEFVCVNVBVOEUDUEUSVOAVBEAUFSUGUHUIUSUT
+    RCZVMVGOUSVQVRVTPVSDAUJQZUTUKTULSUMUSVRVJVFOVSAUNTUSVTVARCVKVIOWAUTUOVAUPUQ
+    UR $.
+
+  ${
+    $d A k n $.  $d F k $.
+    efi4p.1 $e |- F = ( n e. NN0 |-> ( ( ( _i x. A ) ^ n ) / ( ! ` n ) ) ) $.
+    $( Separate out the first four terms of the infinite series expansion of
+       the exponential function of an imaginary number.  (Contributed by Paul
```

Comment:
remove "of an imaginary number" (not needed, and the statement is made for an arbitrary complex number anyway)

Reply:
``( exp ` ( _i x. A ) )`` is what is being expanded in `efi4p`, so it is pure imaginary rather than arbitrary complex. The arbitrary complex one is `ef4p`.

To my ears "separate out" is a natural way to express the idea of listing the first few terms explicitly, then a summation expression for the rest.

Reply:
Yes, it's `( _i x. A )`, but the line reads ``efi4p $p |- ( A e. CC -> ( exp ` ( _i x. A ) ) =`` so it's any complex number.

Reply:
Hmm, that's a bit of a tough one. Although `A` can be any complex number, the normal use case would be where it is real, and in fact both usages currently in set.mm supply `A` which is always real. I guess removing `of an imaginary number` is fine.  I've pushed up a change to do that in set.mm and iset.mm.